src/App.js

```diff
@@ -2,97 +2,101 @@ import React from "react";
   state = {
     result: "",
-    done: false,
-    operate: undefined,
-    originalLastNum: "",
     lastEquation: "",
   };
 
   onClick = (button) => {
+    const { result, lastEquation } = this.state;
+
     const pressedEqual = button.key === "=";
     const openParens = button.key === "(";
     const closeParens = button.key === ")";
-    const pressedDecimal = button.key === ".";
     const numOrOperator = "+-*/0123456789.".includes(button.key);
     const numOrDecimal = "0123456789.".includes(button.key);
     const pressedAC = button.key === "AC";
     const pressedCE = button.key === "CE";
 
+    const getLastChar = (from) => {
+      return from.slice(-1);
+    };
+
     if (numOrDecimal || openParens || closeParens) {
-      if (openParens || closeParens) {
-        this.setState({
-          result: this.state.result + button.key,
-        });
+      this.setState({
+        result: result + button.key,
+      });
+    } else if (numOrOperator) {
+      this.setState({
+        result: changeKeys(result, button.key),
+      });
+    } else if (pressedEqual) {
+      var cleanEquation = result;
+
+      if ("-+*/".includes(getLastChar(result))) {
+        cleanEquation = result.slice(0, -1);
+      }
+
+      cleanEquation = closeOpenParens(cleanEquation);
+      cleanEquation = addMultiplier(cleanEquation);
+
+      let finalResult = 0;
+      var nextEquation = cleanEquation;
+
+      if (lastEquation.length > 0) {
+        const operatorAndNumRegex = /[-+/*]{0,}[0-9]{1,}[.]{0,1}[0-9]*/g;
+        const results = lastEquation.match(operatorAndNumRegex) || [];
+        const lastOperatorAndNum = results[results.length - 1];
+        const lastOperator = lastOperatorAndNum[0];
+        const lastNum = lastOperatorAndNum.substr(1);
+        const roundedResult = (expression) => {
+          const digitAfterComma = 13;
+          return round(evaluate(expression), digitAfterComma);
+        };
+
+        if ("+-/*".includes(lastOperator) && !isNaN(lastNum)) {
+          try {
+            nextEquation = cleanEquation + lastOperator + lastNum;
+            finalResult = roundedResult(nextEquation) + "";
+          } catch (error) {}
+        }
       } else {
-        if (pressedDecimal || (this.state.originalLastNum + "").includes(".")) {
-          var [originalLastNum, operate] = recordLastNum(
-            this.state.result + button.key
-          );
-          this.setState({
-            originalLastNum: originalLastNum,
-            operate: operate,
-            result: this.state.result + button.key,
-            // lastEquation: this.state.result + button.key,
-          });
-        } else {
-          this.setState({
-            //originalLastNum: button.key,
-            originalLastNum: recordLastNum(this.state.result + button.key),
-            result: this.state.result + button.key,
-            // lastEquation: this.state.result + button.key,
-          });
+        const nonNestedParens = /\(([0123456789/*-+.]*)\)/g;
```

Comment:
@grimxreaper lots of changes happening at once in this commit. did you end up referencing the other branch to make this change?

Reply:
Which change? Yes I did at some points, but very minimally and only after many hours of trying to combine these setStates.

Reply:
I'm taking your comment to mean that my comments are not showing up in the commit where I'm seeing it. I'll add the commit message in future comments.

Reply:
I was referring to the commit titled "remove state variables, update pressedEqual functions"

Reply:
Yes it would help, because the comments are showing up on lines of code that are outdated.